### PR TITLE
[datasource] packet_operating_system -> data source

### PR DESCRIPTION
**Pull request by jasmingacic**
Here is the new data source for packet provider.
```
make testacc TESTARGS='-run=TestAccPacketOperatingSystem_Basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccPacketOperatingSystem_Basic -timeout 120m
?       github.com/terraform-providers/terraform-provider-packet        [no test files]
=== RUN   TestAccPacketOperatingSystem_Basic
--- PASS: TestAccPacketOperatingSystem_Basic (4.76s)
PASS
ok      github.com/terraform-providers/terraform-provider-packet/packet 4.788s
```

### CODE REVIEW

**Comment by t0mk** [comment]
hi, can you illustrate a practical use case of the datasource in the markdown doc? So that users understand what's it good for? like in

https://www.terraform.io/docs/providers/do/d/image.html

File: website/docs/d/operating_system.html.markdown [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+layout: "packet"
+page_title: "Packet: opearating_system"
```
Comment (t0mk) [inline website/docs/d/operating_system.html.markdown]: s/opearating/operating

File: website/docs/d/operating_system.html.markdown [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+sidebar_current: "docs-packet-datasource-operating-system"
+description: |-
+  Get an operationg on a Packet Operating System image
```
Comment (t0mk) [inline website/docs/d/operating_system.html.markdown]: description doesn't make sense

File: website/docs/d/operating_system.html.markdown [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+## Argument Reference
+
+ * `distro` - (Optional) Name of the OS distribution.
+ * `name` - (Optional) Name or part of the name of the distribution. Case insensitive.
+ * `provisionable_on` - (Optional) Plan name.
+ * `version` - (Optional) Version of the distribution
+
+## Attributes Reference
+
+ * `operating_system` - Opearting system of a device.
```
Comment (t0mk) [inline website/docs/d/operating_system.html.markdown]: s/Opearting/Operating/

**Comment by jasmingacic** [comment]
@t0mk here you go.

**Comment by t0mk** [comment]
@jasmingacic Thanks! I tested the example and tried a few corner cases, and I confirm it works.

Just one more thing I suggest:
If I would do it, I would add a computed param for slug to the schema
```go
"slug": {
	Type:     schema.TypeString,
	Computed: true,
},
```

I can see that the slug goes to the `Id` of the resource, and another param would be duplicating it, but it might feel more consistent to the Packet users to use the `slug` param in hcl templates (` operating_system = "${data.packet_operating_system.example.slug}"`), instead of the Id, which might wrongly appear like UUID (and there's no UUID in the OS resources in Packet).

It's your PR so I leave that ^ up to you to decide. Let me know if you want to do it, and I will merge or wait accordingly.

**Comment by jasmingacic** [comment]
I've added `slug` as a computed field. So if you think it is ready go ahead and merge it.

**Pull request merged**